### PR TITLE
[Merged by Bors] - feat(Combinatorics/SimpleGraph): vertices are not reachable iff set of walks between them is empty

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
This lemma came up while extending `SimpleGraph.dist` to `ENat`.

Co-authored-by: Kyle Miller <kmill31415@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Connectivity.lean [hunk 1]
```diff
@@ -2001,6 +2001,9 @@ theorem reachable_iff_nonempty_univ {u v : V} :
   Set.nonempty_iff_univ_nonempty
 #align simple_graph.reachable_iff_nonempty_univ SimpleGraph.reachable_iff_nonempty_univ
 
+lemma not_reachable_iff_isEmpty_walk {u v : V} : ¬G.Reachable u v ↔ IsEmpty (G.Walk u v) := by
+   rw [reachable_iff_nonempty_univ, ← Set.nonempty_iff_univ_nonempty, not_nonempty_iff]
```
Comment (IvanRenison) [inline Mathlib/Combinatorics/SimpleGraph/Connectivity.lean:2005]:
```suggestion
  rw [reachable_iff_nonempty_univ, ← Set.nonempty_iff_univ_nonempty, not_nonempty_iff]
```

Comment (IvanRenison) [reply Mathlib/Combinatorics/SimpleGraph/Connectivity.lean:2005]: And maybe this also works?
```suggestion
  rw [Reachable, not_nonempty_iff]
```

Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/Connectivity.lean]: Thank you!

**Comment by IvanRenison** [comment]
maintainer merge

**Review by rwst — APPROVED** [review]
LGTM

**Review by IvanRenison — APPROVED** [review]

**Pull request closed without merging**